fix command line argument order

kunkku opened the pull request:
The mount spec and directory are passed as the first and second
arguments to the helper, followed by the options.

The current implementation works on glibc but fails on POSIX-conforming
C libraries such as musl.

relan commented:
Thanks for the patch!

You mean the POSIXLY_CORRECT case, right?

A user can call mount.exfat-fuse directly and is in full right to put device and mountpoint after other options. Your change breaks this use-case. It also breaks `mount.exfat-fuse -V`.

I'd suggest to mutate argv[] instead (i.e. to do what glibc does) to cover all those cases.

kunkku committed:
allow command line argument order used by mount(8)

The mount spec and directory are passed as the first and second
arguments to the helper, followed by the options.

The current implementation works on glibc but fails on POSIX-conforming
C libraries such as musl.

kunkku commented:
Arbitrary argument order supported now.
